clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1281,17 +1259,11 @@ private void recordTopicFetchMetrics(String topic, int bytes, int records) {
                 Map<String, String> metricTags = Collections.singletonMap("topic", topic.replace('.', '_'));
 
                 bytesFetched = this.metrics.sensor(name);
-                bytesFetched.add(this.metrics.metricName("fetch-size-avg",
-                        this.metricGrpName,
-                        "The average number of bytes fetched per request for topic " + topic,
+                bytesFetched.add(this.metrics.metricInstance(metricsRegistry.topicFetchSizeAvg,
```

Comment:
You're right. There's no programmatic enforcement right now. It will have to be done by convention. Right now, the new "convention" is harder to notice, because most things are still instantiating MetricName's directly. I hope to transition as much as possible over to MetricNameTemplate's in upcoming checkins. I hope this transition won't take too long. After the transition, it'll be easier for people to notice and use the new convention.

I had hoped to the transition in one fell swoop (as in https://github.com/apache/kafka/pull/1202), but that proved to be too much for me to accomplish all at once.

There *might* be a way to programmatically enforce it by doing something with annotations and at runtime compare a MetricName with something in a registry. But that seemed excessively complicated.